### PR TITLE
Update date display & storybook args for banner

### PR DESCRIPTION
**Pull request by bkjohnson**
## Description

Related to https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/198

This updates the `MaintenanceBanner` stories with some new content, and also puts a comma after the formatted day. It also replaces `moment` with `date-fns` _only in `MaintenanceBanner`_. This allows us to properly format the `a.m.`/`p.m.` according to [our own guidelines](https://design.va.gov/content-style-guide/dates-and-numbers#times-and-time-zones).

I also found a bug that _might_ be related to daylight savings time. The displayed time was an hour off from the time that was passed in as a prop. This has been fixed.

## Testing done

- Local Storybook
- Unit testing

### Before banner

image

### During banner

image

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

**Review by cvalarida — APPROVED** [review]
Two things:
1. Does this need any unit tests updated?
    - If not, why not?
2. I see in [your comment here](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/198#issuecomment-777028487) that you advise we move away from `moment` because we can't format the `a.m.` / `p.m.`.
    - This _is_ possible in `moment`—we're doing that on `vets-website` via some fancy `moment` magic.
    - That said, I'm all for adding `date-fns` or something else here to accomplish the job if that'll do the trick.
        - We'll just need to measure the net effect on the library's output.

**Comment by rtwell** [comment]
lgtm!

File: package.json [hunk 1]
```diff
@@ -34,6 +34,8 @@
   },
   "dependencies": {
     "classnames": "^2.2.6",
+    "date-fns": "^2.17.0",
+    "date-fns-tz": "^1.1.1",
```
Comment (bkjohnson) [inline package.json:38]: This is to get the formatting that we want, and also to move away from `moment`, which is in [maintenance mode only](https://momentjs.com/docs/#/-project-status/).

File: src/components/MaintenanceBanner/MaintenanceBanner.jsx [hunk 2]
```diff
@@ -67,18 +80,17 @@ export class MaintenanceBanner extends Component {
...
-    const expiresAtET = expiresAt.clone().subtract(4, 'hours');
-
-    if (startsAt.isSame(expiresAt, 'day')) {
+    if (isSameDay(startsAt, expiresAt)) {
       return (
         <>
           <p>
-            <strong>Date:</strong> {startsAtET.format('dddd MMMM D, YYYY')}
+            <strong>Date:</strong>{' '}
+            {formatEastern(startsAt, 'EEEE, MMMM d, yyyy')}
```
Comment (bkjohnson) [inline src/components/MaintenanceBanner/MaintenanceBanner.jsx:88]: The `format` tokens used by `date-fns` are different than the custom ones that `moment` uses. [See this doc for more info](https://github.com/date-fns/date-fns/blob/master/docs/unicodeTokens.md).

File: src/components/MaintenanceBanner/MaintenanceBanner.stories.jsx [hunk 3]
```diff
@@ -14,34 +14,29 @@ export default {
...
-  warnStartsAt: moment().subtract(30, 'minutes').valueOf(),
+  title: 'Site maintenance',
+  warnTitle: 'Upcoming site maintenance',
+  content:
+    'We’re working on VA.gov right now. If you have trouble signing in or using tools, check back after we’re finished. Thank you for your patience.',
+  warnContent:
+    'We’ll be doing some work on VA.gov. The maintenance will last 1 hour. During that time, you won’t be able to sign in or use tools.',
+  startsAt: new Date(Date.now()),
+  expiresAt: addHours(Date.now(), 1),
+  warnStartsAt: subMinutes(Date.now(), 30),
```
Comment (bkjohnson) [inline src/components/MaintenanceBanner/MaintenanceBanner.stories.jsx:39]: The content changes here are for the original request in @rtwell's issue comment.

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@department-of-veterans-affairs/component-library",
-  "version": "1.1.3",
+  "version": "2.0.0",
```
Comment (bkjohnson) [inline package.json:3]: Breaking change since the `<MaintenanceBanner>` props are expecting `Date` objects instead of `moments`

**Review by bkjohnson — COMMENTED** [review]
I haven't tested this in `vets-website` yet, but based on the build output it looks like I'll have to add `date-fns` and `date-fns-tz` as dependencies there since the built files are still trying to `require` those dependencies.

File: package.json (hunk 4 above)
Comment (cvalarida) [reply package.json:3]: Ooh, good call. Let's be sure to make an announcement about this when we release it.

**Pull request merged**